src/core/gui/PageView.cpp

```diff
@@ -755,8 +755,8 @@ void XojPageView::repaintPage() const { xournal->getRepaintHandler()->repaintPag
 
 void XojPageView::repaintArea(double x1, double y1, double x2, double y2) const {
     double zoom = xournal->getZoom();
-    xournal->getRepaintHandler()->repaintPageArea(this, std::lround(x1 * zoom) - 10, std::lround(y1 * zoom) - 10,
-                                                  std::lround(x2 * zoom) + 20, std::lround(y2 * zoom) + 20);
+    xournal->getRepaintHandler()->repaintPageArea(this, std::lround(x1 * zoom), std::lround(y1 * zoom),
+                                                  std::lround(x2 * zoom), std::lround(y2 * zoom));
```

Comment:
I think some artefacts are due to using `std::lround`. You should use `std::floor` on `x1` and `y1`, but `std::ceil` on `x2` and `y2` (to make sure the integral rectangle contains the double rectangle).

Reply:
```suggestion
    xournal->getRepaintHandler()->repaintPageArea(this, static_cast<int>(std::floor(x1 * zoom)), static_cast<int>(std::floor(y1 * zoom)),
                                                  static_cast<int>(std::ceil(x2 * zoom)), static_cast<int>(std::ceil(y2 * zoom)));
```